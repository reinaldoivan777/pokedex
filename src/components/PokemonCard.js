import React, { Fragment } from "react";
import { Col, Card } from "react-bootstrap";
import { Link } from "react-router-dom";

import Badges from "./common/Badges";

const PokemonCard = ({ pokemon: { name, id, number, image, types } }) => {
  return (
    <Fragment>
      <Col xs={12} sm={4} className="my-3">
        <Link to={`/pokemon/${id}`}>
          <Card className="card-home">
            <Card.Img variant="top" src={image} />
            <Card.Body>
              <Card.Title>{`${number} - ${name}`}</Card.Title>
              {types.length && <Badges datas={types} variant="info" />}
            </Card.Body>
          </Card>
        </Link>
      </Col>
    </Fragment>
  );
};

export default PokemonCard;
