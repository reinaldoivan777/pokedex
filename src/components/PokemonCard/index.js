import React, { Fragment } from "react";
import { Col, Card, Button } from "react-bootstrap";
import { Link } from "react-router-dom";
import { isEmpty } from "lodash";

import Badges from "../common/Badges";

const PokemonCard = ({ pokemon }) => {
  if (!pokemon || isEmpty(pokemon)) return null;
  const { name, id, number, image, types } = pokemon;
  return (
    <Fragment>
      <Col xs={12} sm={4} className="my-3">
        <Link to={`/pokemon/${id}`}>
          <Card className="card-home">
            <Card.Img variant="top" src={image} />
            <Card.Body>
              <Card.Title>{`${number} - ${name}`}</Card.Title>
              {types.length && <Badges datas={types} variant="info" />}
            </Card.Body>
            <div className="overlay">
              <Button className="middle" variant="primary">
                MORE DETAILS
              </Button>
            </div>
          </Card>
        </Link>
      </Col>
    </Fragment>
  );
};

export default PokemonCard;
