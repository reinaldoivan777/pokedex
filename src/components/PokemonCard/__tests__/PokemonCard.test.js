import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

import PokemonCard from "../index";

describe("PokemonCard", () => {
  it("render without error", () => {
    const data = {
      name: "Bulbasaur",
      types: ["Grass", "Poison"],
      id: "UG9rZW1vbjowMDE=",
      number: "001",
      image: "https://img.pokemondb.net/artwork/bulbasaur.jpg"
    };
    renderer.create(
      <Router>
        <PokemonCard pokemon={data} />
      </Router>
    );
  });

  it("empty data", () => {
    renderer.create(
      <Router>
        <PokemonCard pokemon={{}} />
      </Router>
    );
  });

  it("data null", () => {
    renderer.create(
      <Router>
        <PokemonCard pokemon={null} />
      </Router>
    );
  });

  it("snapshot renders", () => {
    const data = {
      name: "Bulbasaur",
      types: ["Grass", "Poison"],
      id: "UG9rZW1vbjowMDE=",
      number: "001",
      image: "https://img.pokemondb.net/artwork/bulbasaur.jpg"
    };
    const component = renderer.create(
      <Router>
        <PokemonCard pokemon={data} />
      </Router>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
