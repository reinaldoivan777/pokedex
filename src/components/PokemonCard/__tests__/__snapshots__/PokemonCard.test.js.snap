// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCard snapshot renders 1`] = `
<div
  className="my-3 col-sm-4 col-12"
>
  <a
    href="/pokemon/UG9rZW1vbjowMDE="
    onClick={[Function]}
  >
    <div
      className="card-home card"
    >
      <img
        className="card-img-top"
        src="https://img.pokemondb.net/artwork/bulbasaur.jpg"
      />
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          001 - Bulbasaur
        </div>
        <span
          className="mr-1 badge badge-info"
        >
          Grass
        </span>
        <span
          className="mr-1 badge badge-info"
        >
          Poison
        </span>
      </div>
    </div>
  </a>
</div>
`;
